lib/cisco_node_utils/interface_channel_group.rb

```diff
@@ -64,31 +64,63 @@ def channel_group
       config_get('interface_channel_group', 'channel_group', @get_args)
     end
 
-    def channel_group=(group)
-      # 'force' is needed by NXOS to handle the case where a port-channel
-      # interface is created prior to the channel-group cli; in which case
-      # the properties of the port-channel interface will be different from
-      # the ethernet interface. 'force' is not needed if the port-channel is
-      # created as a result of the channel-group cli but since it does no
-      # harm we will use it every time.
-      # IOS XR simply ignores 'force'
+    def mode
+      config_get('interface_channel_group', 'mode', @get_args)
+    end
+
+    def feature_lacp?
+      config_get('interface_channel_group', 'feature_lacp')
+    end
+
+    def feature_lacp_set(val)
+      return if feature_lacp? == val
+      config_set('interface_channel_group', 'feature_lacp',
+                 state: val ? '' : 'no')
+    end
+
+    # common setter
+    def channel_group_set(group, mode)
+      feature_lacp_set if mode == 'active' || mode == 'passive'
       if group
         state = ''
         force = 'force'
       else
         state = 'no'
         group = force = ''
       end
-      config_set('interface_channel_group', 'channel_group',
-                 set_args_keys(state: state, group: group, force: force))
+      config_set('interface_channel_group', 'channel_group_set',
+                 set_args_keys(state: state, group: group, force: force,
+                               mode: mode))
     rescue Cisco::CliError => e
-      # Some XR platforms do not support channel-group configuration
-      # on some OS versions. Since this is an OS version difference and not
-      # a platform difference, we can't handle this in the YAML.
       raise unless e.message[/the entered commands do not exist/]
       raise Cisco::UnsupportedError.new('interface', 'channel_group')
     end
 
+    # def channel_group=(group)
```

Comment:
Yeah, this one is a little more complex. For example, if you enter `channel-mode 123 mode on`, the resultant config present in the device is simply `channel-mode 123`.  I modeled using the vPC keepalive setter since it was multiple parameters being based into a single line similar to this type. I'll leave it like the latest commit for now as I was able to get the getters & setter working in IRB on a switch. See below:

```
SB2SF02LF01(config)# int eth1/53
SB2SF02LF01(config-if)# channel-group 100 mode active 
SB2SF02LF01(config-if)# end
SB2SF02LF01# show run int eth1/53

!Command: show running-config interface Ethernet1/53
!Time: Thu Mar  9 19:34:12 2017

version 7.0(3)I4(2)

interface Ethernet1/53
  description I,vPC Peerlink Po100
  switchport
  switchport mode trunk
  spanning-tree port type network
  storm-control broadcast level 90.00
  storm-control multicast level 90.00
  channel-group 100 mode active
  no shutdown

irb(main):008:0> test = Cisco::Node.new()
^[[A=> Node: client:'<Cisco::Client::NXAPI of localhost>' cmd_ref:'CommandReference for 'N9K-C93180YC-EX' (platform:'nexus', data formats:[:nxapi_structured, :cli]) based on 71 files'
irb(main):009:0> test.config_get('interface_channel_group','description',name: 'ethernet1/53')
=> "I,vPC Peerlink Po100"
irb(main):010:0> test.config_get('interface_channel_group','channel_group',name: 'ethernet1/53')
=> 100
irb(main):011:0> test.config_get('interface_channel_group','channel_mode',name: 'ethernet1/53') 
=> "active"
irb(main):006:0> test.config_set('interface_channel_group','channel_group_set',{name: 'ethernet1/53', state: '', force: 'force', group: '200', mode: 'active'})  
=> [{"code"=>"200", "msg"=>"Success", "body"=>{}}, {"code"=>"200", "msg"=>"Success", "body"=>{}}]
---
SB2SF02LF01# show run int eth1/53

!Command: show running-config interface Ethernet1/53
!Time: Thu Mar  9 19:38:40 2017

version 7.0(3)I4(2)

interface Ethernet1/53
  description I,vPC Peerlink Po100
  channel-group 200 mode active
  no shutdown